Source/AliveLibAE/Sound/Midi.cpp

```diff
@@ -867,3 +867,220 @@ EXPORT void CC SND_Restart_4CB0E0()
+            sample->sampleRate = 8000;        // non standard? Doesn't use sampleRate field?
+            sample->loop = sampleRate > 44100; // non-standard?
+            samples.push_back(sample);
+        }
+
+        soundDatFile.close();
+
+        return samples;
+    }
+};
+
+psx::MidiPlayer* player = new psx::MidiPlayer(new AEResourceProvider(), new AESoundSampleParser());
+
+EXPORT void CC SND_StopAll_4CB060()
+{
+    MusicController::EnableMusic_47FE10(FALSE);
+    BackgroundMusic::Stop_4CB000();
+    player->SND_StopAll();
+}
+
+EXPORT void CC SND_Init_4CA1F0()
+{
+    player->SND_Init();
+}
+
+EXPORT void CC SND_Shutdown_4CA280()
+{
+    player->SND_Shutdown();
+}
+
+EXPORT void CC SND_Stop_Channels_Mask_4CA810(u32 bitMask)
+{
+    bitMask;
+    player->SND_Stop_Channels_Mask(bitMask);
+}
+
+EXPORT void SND_Reset_4C9FB0()
+{
+    player->SND_Reset();
+}
+
+EXPORT void CC SND_Load_VABS_4CA350(SoundBlockInfo* pSoundBlockInfo, s32 reverb)
+{
+    pSoundBlockInfo;
+    reverb;
+    player->SND_Load_VABS(reinterpret_cast<psx::SoundBlockInfo*>(pSoundBlockInfo), reverb);
+}
+
+EXPORT s32 CC SND_4CA5D0(s32 program, s32 vabId, s32 note, s16 vol, s16 min, s16 max)
+{
+    return player->SND(program, vabId, note, vol, min, max);
+}
+
+EXPORT void CC SND_Restart_4CB0E0()
+{
+    MusicController::EnableMusic_47FE10(TRUE);
+    BackgroundMusic::Play_4CB030();
+    Start_Sounds_For_Objects_In_Near_Cameras_4CBB60();
+    player->SND_Restart();
+}
+
+EXPORT void CC SND_Load_Seqs_4CAED0(OpenSeqHandle* pSeqTable, const char_type* bsqFileName)
+{
+    player->SND_Load_Seqs(reinterpret_cast<psx::OpenSeqHandle*>(pSeqTable), bsqFileName);
+}
+
+EXPORT void CC SND_SEQ_Stop_4CAE60(u16 idx)
+{
+    player->SND_SEQ_Stop(idx);
+}
+
+EXPORT s8 CC SND_Seq_Table_Valid_4CAFE0()
+{
+    return player->SND_Seq_Table_Valid();
+}
+
+EXPORT s16 CC SND_SEQ_PlaySeq_4CA960(u16 idx, s16 repeatCount, s16 bDontStop)
+{
+    return player->SND_SEQ_PlaySeq(idx, repeatCount, bDontStop);
+}
+
+EXPORT void CC SND_SEQ_SetVol_4CAD20(s32 idx, s16 volLeft, s16 volRight)
+{
+    player->SND_SEQ_SetVol(idx, volLeft, volRight);
+}
+
+EXPORT s16 CC SND_SEQ_Play_4CAB10(u16 idx, s16 repeatCount, s16 volLeft, s16 volRight)
+{
+    return player->SND_SEQ_Play(idx, repeatCount, volLeft, volRight);
+}
+
+EXPORT s32 CC SND_SsIsEos_DeInlined_4CACD0(u16 idx)
+{
+    return player->SND_SsIsEos_DeInlined(idx);
+}
+
+EXPORT s32 CC SFX_SfxDefinition_Play_4CA700(const SfxDefinition* sfxDef, s16 volLeft, s16 volRight, s16 pitch_min, s16 pitch_max)
+{
+    return player->SFX_SfxDefinition_Play(reinterpret_cast<psx::SfxDefinition*>(const_cast<SfxDefinition*>(sfxDef)), volLeft, volRight, pitch_min, pitch_max);
+}
+
+EXPORT s32 CC SFX_SfxDefinition_Play_4CA420(const SfxDefinition* sfxDef, s16 volume, s16 pitch_min, s16 pitch_max)
+{
+    return player->SFX_SfxDefinition_Play(reinterpret_cast<psx::SfxDefinition*>(const_cast<SfxDefinition*>(sfxDef)), volume, pitch_min, pitch_max);
+} 
+
+#endif
```

Comment:
It's good practice to end the source code files with an endline at the end.  Also, not having one in an .hpp file can cause problems with including multiple ones one after the other.

Seems multiple files miss it here.